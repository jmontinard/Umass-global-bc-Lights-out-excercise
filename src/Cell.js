import React from "react";
import { Box } from "@mui/material";
import "./Cell.css";

/** A single cell on the board.
 *
 * This has no state --- just two props:
 *
 * - flipCellsAroundMe: a function rec'd from the board which flips this
 *      cell and the cells around of it
 *
 * - isLit: boolean, is this cell lit?
 *
 * This handles clicks --- by calling flipCellsAroundMe
 *
 **/

// function Cell({ flipCellsAroundMe, isLit }) {
//   const classes = `Cell ${isLit ? "Cell-lit" : ""}`;
//   return <td className={classes} onClick={flipCellsAroundMe} />;
// }

// export default Cell;


function Cell({ flipCellsAroundMe, isLit }) {
  const cellStyle = {
    height: "100px",
    width: "100px",
    backgroundColor: isLit ? "white" : "darkgray",
    cursor: "pointer",
    borderRadius: "8px", // Add rounded corners
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
  };

  return <Box style={cellStyle} onClick={flipCellsAroundMe} />;
}

export default Cell;