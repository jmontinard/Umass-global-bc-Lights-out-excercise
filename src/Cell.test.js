// import React from "react";
// import { render, fireEvent } from "@testing-library/react";
// import Cell from "./Cell";

// test("renders a Cell properly", () => {
//   const { container } = render(<Cell />);
//   expect(container.firstChild).toBeInTheDocument();

//   // Snapshot test
//   expect(container).toMatchSnapshot();
// });

// test("handles click event properly", () => {
//   const mockClickHandler = jest.fn();
//   const { container } = render(<Cell flipCellsAroundMe={mockClickHandler} />);

//   // Find the cell (you may need to adjust this based on your component structure)
//   const cell = container.querySelector(".MuiBox-root");

//   // Click the cell
//   fireEvent.click(cell);

//   // Add assertions to ensure that the click handler is called
//   expect(mockClickHandler).toHaveBeenCalledTimes(1);

//   // Snapshot test
//   expect(container).toMatchSnapshot();
// });

// // Smoke test
// test("renders without errors", () => {
//   const mockClickHandler = jest.fn();
//   render(<Cell flipCellsAroundMe={mockClickHandler} />);
// });

import React from 'react';
import { render } from '@testing-library/react';
import Cell from './Cell';
import renderer from 'react-test-renderer';

// Smoke Test
it('renders Cell without crashing', () => {
  render(<Cell isLit={false} flipCellsAroundMe={() => {}} />);
});

// Snapshot Test
it('matches snapshot for lit cell', () => {
  const tree = renderer.create(<Cell isLit={true} flipCellsAroundMe={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('matches snapshot for unlit cell', () => {
  const tree = renderer.create(<Cell isLit={false} flipCellsAroundMe={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Test rendering a Cell properly
it('renders lit Cell correctly', () => {
  const { container } = render(<Cell isLit={true} flipCellsAroundMe={() => {}} />);
  expect(container.firstChild).toHaveStyle(`backgroundColor: white`);
});

it('renders unlit Cell correctly', () => {
  const { container } = render(<Cell isLit={false} flipCellsAroundMe={() => {}} />);
  expect(container.firstChild).toHaveStyle(`backgroundColor: darkgray`);
});
