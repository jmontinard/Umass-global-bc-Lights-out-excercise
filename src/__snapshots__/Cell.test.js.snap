// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for lit cell 1`] = `
<div
  className="MuiBox-root css-0"
  onClick={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": "8px",
      "cursor": "pointer",
      "display": "flex",
      "height": "100px",
      "justifyContent": "center",
      "width": "100px",
    }
  }
/>
`;

exports[`matches snapshot for unlit cell 1`] = `
<div
  className="MuiBox-root css-0"
  onClick={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "darkgray",
      "borderRadius": "8px",
      "cursor": "pointer",
      "display": "flex",
      "height": "100px",
      "justifyContent": "center",
      "width": "100px",
    }
  }
/>
`;
